apps/balancer-tools/src/app/apr/api/route.ts

```diff
@@ -232,6 +233,29 @@ const fetchDataForRoundId = async (
   };
 };
 
+function validateSearchParams(poolId: string | null, roundId: string | null) {
+  if ((!poolId || poolId === null) && (!roundId || roundId === null)) {
+    throw new Error("no roundId or poolId provided");
+  }
+  if (poolId) {
+    if (
+      !POOLS_WITH_LIVE_GAUGES.find(
+        (g) => g.id.toLowerCase() === poolId.toLowerCase(),
+      )
```

Comment:
we should centralize whatever logic depends on gauges and prepare to remove it... soon we'll need to support pools that are not in the veBAL system/don't have gauges (mostly in other chains). I flagged this multiple times already and no actionables here but let's just keep it in mind not to create too many dependencies around something that we'll need to rework on